### PR TITLE
Support serializing a std::vector as an array.

### PR DESCRIPTION
**Pull request by richiware**
This PR adds API for encoding/decoding a `std::vector` as an array. This API is needed for Fast DDS 3.0 Dynamic Types.

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/344/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/256/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/345/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/346/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/258/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/347/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/348/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/260/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/349/)

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Changes LGTM but, do we need to extend the test suite to include test cases that check this new feature?

File: fastcdr.repos [hunk 1]
```diff
@@ -3,3 +3,7 @@ repositories:
         type: git
         url: https://github.com/eProsima/Fast-CDR.git
         version: master
+    googletest-distribution:
+        type: git
+        url: https://github.com/google/googletest.git
+        version: release-1.11.0
```
Comment (JLBuenoLopez-eProsima) [inline fastcdr.repos:9]: This probably should not be included in the PR

File: test/cdr/array_as_std_vector.cpp [hunk 2]
```diff
@@ -0,0 +1,442 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (JLBuenoLopez-eProsima) [inline test/cdr/array_as_std_vector.cpp:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/cdr/array_as_std_vector.cpp [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        fval, ival, ival, ival  // ULong
+    };
+    //}
+
+    EncodingAlgorithmFlag encoding = std::get<0>(GetParam());
+    Cdr::Endianness endianness = std::get<1>(GetParam());
+
+    serialize_array(expected_streams, encoding, endianness, array_value);
+
+    serialize_array(expected_streams, encoding, endianness, array_value);
```
Comment (JLBuenoLopez-eProsima) [inline test/cdr/array_as_std_vector.cpp:268]: This instruction is repeated. In the basic_types tests both the upper and low API is tested. Here only one of the APIs is implemented.

File: test/cdr/array_as_std_vector.cpp [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        ival, fval
+    };
+    //}
+
+    EncodingAlgorithmFlag encoding = std::get<0>(GetParam());
+    Cdr::Endianness endianness = std::get<1>(GetParam());
+
+    serialize_array(expected_streams, encoding, endianness, array_value);
+
+    serialize_array(expected_streams, encoding, endianness, array_value);
```
Comment (JLBuenoLopez-eProsima) [inline test/cdr/array_as_std_vector.cpp:425]: Repeated

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/373/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/278/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/353/)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI

File: fastcdr.repos (hunk 1 above)
Comment (richiware) [reply fastcdr.repos]: Done in 7a6d9ad

File: test/cdr/array_as_std_vector.cpp (hunk 2 above)
Comment (richiware) [reply test/cdr/array_as_std_vector.cpp]: Done in 7a6d9ad

File: test/cdr/array_as_std_vector.cpp (hunk 3 above)
Comment (richiware) [reply test/cdr/array_as_std_vector.cpp]: Done in 7a6d9ad

File: test/cdr/array_as_std_vector.cpp (hunk 4 above)
Comment (richiware) [reply test/cdr/array_as_std_vector.cpp]: Done in 7a6d9ad

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/377/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/280/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/357/)
